src/Bridge/Doctrine/Orm/Util/IdentifierManagerTrait.php

```diff
@@ -77,7 +79,11 @@ private function normalizeIdentifiers($id, ObjectManager $manager, string $resou
             $doctrineTypeName = $doctrineClassMetadata->getTypeOfField($propertyName);
 
             if ($isOrm && null !== $doctrineTypeName && DBALType::hasType($doctrineTypeName)) {
-                $identifier = DBALType::getType($doctrineTypeName)->convertToPHPValue($identifier, $platform);
+                try {
+                    $identifier = DBALType::getType($doctrineTypeName)->convertToPHPValue($identifier, $platform);
+                } catch (ConversionException $e) {
+                    throw new InvalidArgumentException(sprintf('Invalid identifier "%s" value provided.', $propertyName));
```

Comment:
We should keep the previous exception as done https://github.com/api-platform/core/pull/1304/files. Not sure about the use case though.